tests-functional/tests/test_add_account.py

```diff
@@ -0,0 +1,148 @@
+        add_account_response = self.account.accounts_service.add_account(self.account_data)
+        self.account.verify_json_schema(add_account_response, method="accounts_addAccount")
+        assert "error" not in add_account_response
+
+        # After adding the account check that the get accounts will retrieve the new account
+        accounts_response = self.account.accounts_service.get_accounts()
+        accounts = accounts_response.get("result", [])
+        self.check_new_account_was_created(accounts, self.account_data)
+
+    def test_add_seed_account(self):
+        self.account_data["key-uid"] = self.get_keypair_key_uid()
+        self.account_data["type"] = "seed"
+        add_account_response = self.account.accounts_service.add_account(self.account_data)
+        self.account.verify_json_schema(add_account_response, method="accounts_addAccount")
+        assert "error" not in add_account_response
+
+    def test_delete_account(self):
+        self.account_data["type"] = "watch"
+        add_account_response = self.account.accounts_service.add_account(self.account_data)
+        self.account.verify_json_schema(add_account_response, method="accounts_addAccount")
+        assert "error" not in add_account_response
+
+        accounts_response = self.account.accounts_service.get_accounts()
+        assert len(accounts_response.get("result", [])) == 3
+
+        # Delete the account
+        delete_response = self.account.accounts_service.delete_account(self.account_data["address"])
+        assert "error" not in delete_response
+        self.account.verify_json_schema(delete_response, method="accounts_deleteAccount")
+
+        accounts_response = self.account.accounts_service.get_accounts()
+        assert len(accounts_response.get("result", [])) == 2
+
+    def get_account_keypairs(self):
+        keypairs_response = self.account.accounts_service.get_account_keypairs()
+        keypairs = keypairs_response.get("result", [])
+        assert len(keypairs) > 0
+        return keypairs
+
+    def get_keypair_key_uid(self):
+        keypairs = self.get_account_keypairs()
+        first_keypair = keypairs[0]
+        return first_keypair.get("key-uid")
+
+    def check_new_account_was_created(self, accounts, account_data):
+        mismatch_details = ""
+        for acc in accounts:
+            mismatches = []
+            for key, value in account_data.items():
+                if key not in acc:
+                    mismatches.append(f"Missing key: {key}")
+                    continue
+                actual_value = acc[key]
+                if key == "address":
+                    if value.lower() != actual_value.lower():
+                        mismatches.append(f"Mismatch in key '{key}': expected '{value}', got '{actual_value}'")
+                else:
+                    if value != actual_value:
+                        mismatches.append(f"Mismatch in key '{key}': expected '{value}', got '{actual_value}'")
```

Comment:
This allows to avoid code duplication
```suggestion
                if key == "address" and value.lower() == actual_value.lower():
                    continue
                if value == actual_value:
                    continue
                mismatches.append(f"Mismatch in key '{key}': expected '{value}', got '{actual_value}'")
```